srcpkgs/genie/template

```diff
@@ -0,0 +1,48 @@
+# Template file for 'genie'
+pkgname=genie
+version=1141
+revision=1
+hostmakedepends="git"
+short_desc="Project generator tool using Lua"
+maintainer="a dinosaur <nick@a-dinosaur.com>"
+license="BSD-3-Clause"
+homepage="https://github.com/bkaradzic/GENie"
+_commit=44918162588e56512ddac6264b08ca6ba4e67468
+
+do_fetch() {
+	git clone "${homepage}.git" "${pkgname}-${version}"
+	cd "${pkgname}-${version}"
+	git reset --hard "${_commit}"
+}
```

Comment:
grep for `githash` to see how to fetch a commit from a github repo

Reply:
@Piraty I changed the convention to match what you were referring to, or did you also mean replacing the usage of Git with just getting the commit directly from GitHub?

The reason I'm using Git is because [these](https://github.com/bkaradzic/GENie/blob/44918162588e56512ddac6264b08ca6ba4e67468/scripts/release.lua#L18) [lines](https://github.com/bkaradzic/GENie/blob/44918162588e56512ddac6264b08ca6ba4e67468/scripts/release.lua#L21) use `git` to glean information about the version, which fails without an actual Git repository.

I could try patching these lines myself because they're just getting strings that I already know about in the template, and then the hostmakedepends=git could be omitted entirely in favour of just fetching the commit directly. Which would be preferred?

Reply:
1. a small hint about git requirement would have made it more obvious
2. you can write `version.h` (as defined [here](https://github.com/bkaradzic/GENie/blob/44918162588e56512ddac6264b08ca6ba4e67468/scripts/release.lua#L18-L27)) on your own with the information available in the template, as well as maybe even skip running that release script entirely and mimic what does in `pre_configure()`. or just patch out the lines calling `git`, whatever you like

Reply:
I did mention it in my PR comment, it should be more obvious now from the template (the git requirement is gone anyway).
Rewriting the whole script would be absurd when the issue can be fixed with patches & sed, so I've just opted for that.